os/kernel/semaphore/sem_wait.c

```diff
@@ -117,10 +120,16 @@ int sem_wait(FAR sem_t *sem)
 	FAR struct tcb_s *rtcb = this_task();
 	irqstate_t saved_state;
 	int ret = ERROR;
-
 	/* This API should not be called from interrupt handlers */
+#if defined(CONFIG_FS_ROMFS) && defined(CONFIG_FRAME_POINTER)
```

Comment:
How call this in assert?

Reply:
As I mentioned in the commit message, I am parsing the System.map file using fopen , fseek etc. 
I have used this approach so that, we don't need to waste RAM memory to save large number of symbols. These file apis make use of semaphore internally. and That's the reason i have used this workaround.

Reply:
This is in assert status. Is using fopen, fseek with sem available always? I think you use too high level APIs.